src/edwards/curve.rs

```diff
@@ -81,6 +84,49 @@ impl EllipticCurve for Curve {
             z: self.f.one(),
         })
     }
+    // based on https://tools.ietf.org/html/rfc8032#section-5.2.3
+    fn deserialize(&self, buf: &[u8]) -> Result<Self::Point,Error> {
+        // step 1
+        let last_byte = buf.len()-1;
+        let x_0 = (buf[last_byte]>>7)&0x01;
+        let mut y_bytes = buf.to_vec();
+        y_bytes[last_byte] = y_bytes[last_byte]&127; // clear msb
+        let y_zz = BigInt::from_bytes_le(Sign::Plus, &y_bytes);
+        let p = self.f.get_modulus();
+        if y_zz > p {
+            return Err(Error::new(ErrorKind::Other, "Invalid y value chosen"));
+        }
+        let y = self.f.elt(y_zz);
+
+        // step 2
+        let yy = &y*&y;
+        let minus_one = -self.f.one();
+        let u = &yy + &minus_one;
+        // v is computed differently for c25519 and c448
```

Comment:
Not really. the formula to recover x  is:
`x =\sqrt{ \frac{ y^2 - 1 }{ dy^2 - a } }`
where `a` and `d` are curve parameters